lib/hexdocs/queue.ex

```diff
@@ -66,7 +66,8 @@ defmodule Hexdocs.Queue do
         version = Version.parse!(version)
         all_versions = all_versions(repository, package)
         Hexdocs.Bucket.upload(repository, package, version, all_versions, files)
-        update_sitemap(repository, key)
+        update_index_sitemap(repository, key)
+        update_package_sitemap(repository, key, package, files)
```

Comment:
Instead of a separate step to upload the package sitemap, add it to the list of files above so that it is uploaded with `Hexdocs.Bucket.upload`. That should simplify things and we can remove some code.

Reply:
@ericmj I agree that uploading all files at the same time would be preferred but I struggled with coming up with a good solution. That's because the index sitemap is on the top-level, the package sitemap is on the package unversioned level, docs_config.js too, and all the package files are versioned/unversioned. Maybe we can ship it as is so search engines can start indexing, otherwise maybe we could pair on refactoring this first?

Reply:
there's one more thing we need to do, we should only update the package sitemap (and thus index sitemap too) when we're at the _latest version_, let me give it a try.